pkg/hooks/manager.go

```diff
@@ -196,18 +196,30 @@ func (m *Manager) OnDefineDomain(domainSpec *virtwrapApi.DomainSpec, vmi *v1.Vir
 				}
 				defer conn.Close()
 
-				client := hooksV1alpha1.NewCallbacksClient(conn)
-
 				ctx, cancel := context.WithTimeout(context.Background(), time.Minute)
 				defer cancel()
-				result, err := client.OnDefineDomain(ctx, &hooksV1alpha1.OnDefineDomainParams{
-					DomainXML: domainSpecXML,
-					Vmi:       vmiJSON,
-				})
-				if err != nil {
-					return "", err
+
+				if callback.Version == hooksV1alpha1.Version {
+					client := hooksV1alpha1.NewCallbacksClient(conn)
+					result, err := client.OnDefineDomain(ctx, &hooksV1alpha1.OnDefineDomainParams{
+						DomainXML: domainSpecXML,
+						Vmi:       vmiJSON,
+					})
+					if err != nil {
+						return "", err
+					}
+					domainSpecXML = result.GetDomainXML()
+				} else {
```

Comment:
Do an else if here and check for v2 so the final else/panic will trigger if something bad did happen.